crates/sui-network/src/state_sync/mod.rs

```diff
@@ -278,9 +314,19 @@ impl Iterator for PeerBalancer {
             let idx =
                 rand::thread_rng().gen_range(0..std::cmp::min(SELECTION_WINDOW, self.peers.len()));
             let (peer, info) = self.peers.remove(idx).unwrap();
-
-            if info.height >= self.requested_checkpoint.unwrap_or(0) {
-                return Some(StateSyncClient::new(peer));
+            let requested_checkpoint = self.requested_checkpoint.unwrap_or(0);
+            // Summary will never be pruned
```

Comment:
nit - move this comment down two lines, since it only applies to the summary arm of the match and not the content arm?